viewer/packages/rendering/src/GeometryBatchingManager.ts

```diff
@@ -107,18 +103,51 @@ export class GeometryBatchingManager {
     if (bufferIsReallocated) {
       this.reallocateBufferGeometry(mesh, defragBuffer);
     } else {
-      this.updateInstanceAttributes(mesh);
+      this.updateInstanceAttributes(mesh, updateRange);
     }
 
     mesh.count += instanceCount;
   }
 
-  private updateInstanceAttributes(mesh: THREE.InstancedMesh<THREE.BufferGeometry, THREE.Material | THREE.Material[]>) {
+  private updateInstanceAttributes(
+    mesh: THREE.InstancedMesh<THREE.BufferGeometry, THREE.Material | THREE.Material[]>,
+    updateRange: {
+      byteOffset: number;
+      byteCount: number;
+    }
+  ) {
     this.getAttributes(mesh.geometry, THREE.InterleavedBufferAttribute).forEach(namedAttribute => {
-      namedAttribute.attribute.data.needsUpdate = true;
+      const attribute = namedAttribute.attribute;
+      this.setUpdateRange(attribute, updateRange);
+      attribute.data.needsUpdate = true;
     });
   }
 
+  private setUpdateRange(
+    attribute: THREE.InterleavedBufferAttribute,
+    updateRange: {
+      byteOffset: number;
+      byteCount: number;
+    }
+  ) {
+    const typeSize = (attribute.data.array as TypedArray).BYTES_PER_ELEMENT;
+    const newOffset = updateRange.byteOffset / typeSize;
+    const newCount = updateRange.byteCount / typeSize;
+
+    const { offset: oldOffset, count: oldCount } = attribute.data.updateRange;
+
+    if (oldCount === -1) {
+      attribute.data.updateRange = { offset: newOffset, count: newCount };
+      attribute.data.needsUpdate = true;
+      return;
+    }
+
+    const offset = Math.min(oldOffset, newOffset);
+    const count = Math.max(oldOffset + oldCount, newOffset + newCount) - offset;
+
+    attribute.data.updateRange = { offset, count };
```

Comment:
The math looks correct, but I'm just wondering: How does `updateRange` work for `InterleavedBufferAttribute`? Does it update the portion of the attribute you specified, plus everything in-between that resides in the same buffer, which may belong to other attributes?